examples/using-file-bind/main_test.go

```diff
@@ -2,19 +2,28 @@ package main
 
 import (
 	"bytes"
+	"fmt"
 	"io"
 	"mime/multipart"
 	"net/http"
 	"os"
+	"strconv"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gofr.dev/pkg/gofr/testutil"
 )
 
 func TestMain_BindError(t *testing.T) {
-	const host = "http://localhost:8300"
+	httpPort := testutil.GetFreePort(t)
+	t.Setenv("HTTP_PORT", strconv.Itoa(httpPort))
+	host := fmt.Sprint("http://localhost:", httpPort)
+
+	port := testutil.GetFreePort(t)
+	t.Setenv("METRICS_PORT", strconv.Itoa(port))
```

Comment:
Not for this PR, but what do you guys think about adding an helper for setting up all this?

It could set up the ports, the env and return a struct, the port and url and host and metrics could be getter on this struct.

I can open an issue for this and suggest a PR, or it could be addressed by anyone if someone is interested.

Reply:
I can also suggest something using TestMain that could help you to make tests even cleaner. But the things will be hidden from sight and somehow magical. So I don't think TestMain should be used for things in examples folder. It's important that code can be easily copy pasted by people who doesn't understand how stuffs work.

But test main could be an hammer for a scratch, maybe a simple NewT(*testing.T) that set up the ports via env and return an app like New() does

Reply:
Sure, we can handle that in another PR.

Reply:
I created 
- https://github.com/gofr-dev/gofr/issues/1352